src/main/java/org/zephyrsoft/sdb2/gui/MainWindow.java

```diff
@@ -1765,28 +1777,103 @@ public void focusLost(FocusEvent e) {
 		
+		JLabel lblDrumNotes = new JLabel("Drum notes");
+		GridBagConstraints gbcLblDrumNotes = new GridBagConstraints();
+		gbcLblDrumNotes.fill = GridBagConstraints.HORIZONTAL;
+		gbcLblDrumNotes.gridwidth = 1;
+		gbcLblDrumNotes.insets = new Insets(0, 0, 5, 5);
+		gbcLblDrumNotes.gridx = 1;
+		gbcLblDrumNotes.gridy = 8;
+		panelEdit.add(lblDrumNotes, gbcLblDrumNotes);
+		
+		JScrollPane scrollPaneDrumNotes = new JScrollPane();
+		GridBagConstraints gbcScrollPaneDrumNotes = new GridBagConstraints();
+		gbcScrollPaneDrumNotes.gridheight = 2;
+		gbcScrollPaneDrumNotes.weighty = 1.0;
+		gbcScrollPaneDrumNotes.fill = GridBagConstraints.BOTH;
+		gbcScrollPaneDrumNotes.gridwidth = 1;
+		gbcScrollPaneDrumNotes.insets = new Insets(0, 0, 0, 5);
+		gbcScrollPaneDrumNotes.gridx = 1;
+		gbcScrollPaneDrumNotes.gridy = 9;
+		panelEdit.add(scrollPaneDrumNotes, gbcScrollPaneDrumNotes);
+		
+		editorDrumNotes = new JEditorPane();
+		editorDrumNotes.setFont(new Font("Monospaced", editorDrumNotes.getFont().getStyle(),
+			editorDrumNotes.getFont().getSize()));
+		scrollPaneDrumNotes.setViewportView(editorDrumNotes);
+		editorDrumNotes.setBackground(Color.WHITE);
+		
+		JLabel lblSongNotes = new JLabel("Song Notes (not shown in presentation)");
+		GridBagConstraints gbcLblSongNotes = new GridBagConstraints();
+		gbcLblSongNotes.fill = GridBagConstraints.HORIZONTAL;
+		gbcLblSongNotes.gridwidth = 1;
+		gbcLblSongNotes.insets = new Insets(0, 0, 5, 5);
+		gbcLblSongNotes.gridx = 2;
+		gbcLblSongNotes.gridy = 8;
+		panelEdit.add(lblSongNotes, gbcLblSongNotes);
+		
+		JScrollPane scrollPaneSongNotes = new JScrollPane();
+		GridBagConstraints gbcScrollPaneSongNotes = new GridBagConstraints();
+		gbcScrollPaneSongNotes.gridheight = 2;
+		gbcScrollPaneSongNotes.weighty = 1.0;
+		gbcScrollPaneSongNotes.fill = GridBagConstraints.BOTH;
+		gbcScrollPaneSongNotes.gridwidth = 1;
+		gbcScrollPaneSongNotes.insets = new Insets(0, 0, 0, 5);
+		gbcScrollPaneSongNotes.gridx = 2;
+		gbcScrollPaneSongNotes.gridy = 9;
+		panelEdit.add(scrollPaneSongNotes, gbcScrollPaneSongNotes);
+		
+		editorSongNotes = new JEditorPane();
+		scrollPaneSongNotes.setViewportView(editorSongNotes);
+		editorSongNotes.setBackground(Color.WHITE);
+		
+		for (JComponent v : new JComponent[] {
+			textFieldTonality,
+			textFieldAdditionalCopyrightNotes,
+			textFieldAuthorText,
+			textFieldAuthorTranslation,
+			textFieldComposer,
+			textFieldPublisher,
+			textFieldTempo,
+			textFieldTitle,
+			(JTextField) comboBoxLanguage.getEditor().getEditorComponent(),
+			editorLyrics,
+			editorChordSequence,
+			editorSongNotes,
+			editorDrumNotes
+		}) {
+			v.addFocusListener(new FocusListener() {
+				
+				@Override
+				public void focusLost(FocusEvent e) {
+					try {
+						// saveSong();
+						saveSongWithoutChangingGUI();
+					} catch (Throwable ex) {
+						handleError(ex);
+					}
+				}
+				
+				@Override
+				public void focusGained(FocusEvent e) {
+					// TODO
```

Comment:
Is here still something to do? If not, `FocusAdapter` might be more fitting if we only need to handle focus-lost events (which I suspect).

Reply:
Yes, good idea.